exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Query.kt

```diff
@@ -444,7 +444,13 @@ open class Query(override var set: FieldSet, where: Op<Boolean>?) : AbstractQuer
                 adjustSelect {
                     select(
                         originalSet.fields.map {
-                            it as? ExpressionAlias<*> ?: ((it as? Column<*>)?.makeAlias() ?: it.alias("exp${expInx++}"))
+                            it as? ExpressionWithColumnTypeAlias<*>
```

Comment:
It could be more readable with `when` variant:

```kotlin
when (it) {
                                is ExpressionWithColumnTypeAlias<*> -> it
                                is ExpressionAlias<*> -> it
                                else -> (it as? Column<*>)?.makeAlias()
                                    ?: (it as? ExpressionWithColumnType<*>)?.alias("exp${expInx++}")
                                    ?: it.alias("exp${expInx++}")
                            }
```